pkg/provider/azure_instances.go

```diff
@@ -298,7 +302,7 @@ func (az *Cloud) InstanceShutdown(ctx context.Context, node *v1.Node) (bool, err
 		providerID, err = cloudprovider.GetInstanceProviderID(ctx, az, types.NodeName(node.Name))
 		if err != nil {
 			// Returns false, so the controller manager will continue to check InstanceExistsByProviderID().
-			if strings.Contains(err.Error(), cloudprovider.InstanceNotFound.Error()) {
+			if errors.Is(err, cloudprovider.InstanceNotFound) {
```

Comment:
Could you also add the same check for L314?

Reply:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_instances.go#L243-L287
L314 check is unnecessary, too